### PR TITLE
[refactor] #3363: Genesis API

### LINKED ISSUE #3363: Genesis API
I'm not at all a fan of this API. I'd rather we had a similarly named method in two different traits. The user is left guessing as to what `<true>` means in this context. It was slightly better when we had an enum, but now I'm convinced we really need two trait extension methods and only one in scope.

_Originally posted by @appetrosyan in https://github.com/hyperledger/iroha/pull/3356#discussion_r1161496792_

### PR DESCRIPTION
**Pull request by ilchu**
## Description

* Introduce sealed traits for block revalidation and accepting transactions
* Make wrappers around those two functions for public use

### Linked issue

Closes #3363 

### Benefits
More idiomatic and self-descriptive usage.

### Checklist

- [x] I've read `CONTRIBUTING.md`
- [x] I've used the standard signed-off commit format (or will squash just before merging)
- [x] All applicable CI checks pass (or I promised to make them pass later)
- [ ] (optional) I've written unit tests for the code changes
- [ ] I replied to all comments after code review, marking all implemented changes with thumbs up

### CODE REVIEW

File: data_model/src/transaction.rs [hunk 1]
```diff
@@ -141,6 +142,60 @@ pub trait Sign {
     ) -> Result<SignedTransaction, iroha_crypto::error::Error>;
 }
 
+/// Sealed trait for accepting transactions. Downstream users
+/// should only use its extension traits [`InGenesis`] and
+/// [`InBlock`] according to the use case.
+#[sealed]
+#[cfg(feature = "std")]
+pub trait Accept: Sized {
```
Comment (mversic) [inline data_model/src/transaction.rs:150]:
```suggestion
#[cfg(feature = "transparent-api")]
pub trait Accept: Sized {
```

**Review by mversic — APPROVED** [review]

**Review by appetrosyan — APPROVED** [review]
Reads much better. Glad you chose to go with a sealed trait.

File: core/src/sumeragi/message.rs [hunk 2]
```diff
@@ -169,7 +146,7 @@ impl From<VersionedCommittedBlock> for BlockCommitted {
 #[non_exhaustive]
 pub struct BlockSyncUpdate {
     /// The corresponding block.
-    block: VersionedCommittedBlock,
+    pub block: VersionedCommittedBlock,
```
Comment (Erigara) [inline core/src/sumeragi/message.rs:149]: This field was private on purpose, to always be sure that block will be revalidated before usage. 
However i think we can return to this in separate PR.

**Review by Erigara — DISMISSED** [review]
I think merge commit accidentally happened.

File: core/src/sumeragi/message.rs (hunk 2 above)
Comment (ilchu) [reply core/src/sumeragi/message.rs:149]: Created #3492.

**Review by mversic — APPROVED** [review]

**Review by mversic — DISMISSED** [review]

**Review by mversic — DISMISSED** [review]

**Review by Erigara — APPROVED** [review]

**Review by mversic — APPROVED** [review]

**Pull request merged**